src/compiler/checker.ts

```diff
@@ -17175,7 +17175,8 @@ namespace ts {
 
             let errorInfo: DiagnosticMessageChain | undefined;
             let relatedInfo: [DiagnosticRelatedInformation, ...DiagnosticRelatedInformation[]] | undefined;
-            let maybeKeys: string[];
+            let maybeKeyList: string[];
+            let maybeKeySet: Set<string>;

```

Comment:
Suggestion. You can just just Map<string, number> where number is index mayBeList index and then you dont need array.

Reply:
I don't think that's true because of the way the set has to be "popped" down to `maybeStart`.

Reply:
Yeah, we'd have to traverse the whole thing every time--which we really don't want to do.